### ISSUE TITLE
Unexpected errors if Flow types were defined

### ISSUE THREAD

**Issue by ifedyukin** [yannickcr/eslint-plugin-react#2173]
Hello!
I've updated from eslint-plugin-react@7.7.0 to eslint-plugin-react@7.12.4 and I got some new specific errors. I got unexpected `react/no-unused-prop-types` error on 5 and 6 lines in the next code listing.
```javascript
// @flow
import React from 'react';

export type ChildComponentProps = {
    value: string,
    name: string
};

export const ParentComponent = ({arrayOfSmth}: Object) => {
    return (
        <div>
            {arrayOfSmth.map(({value, name}: ChildComponentProps) => {
                return (
                    <span>
                        {name ? value : <Text>{name}</Text>}
                    </span>
                );
            })}
        </div>
    );
};
```

```
5:12  error    'value' PropType is defined but prop is never used  react/no-unused-prop-types
6:11  error    'name' PropType is defined but prop is never used   react/no-unused-prop-types
```
Is it expected behavior for my case?

**Comment by ljharb** [comment]
That’s very strange indeed, seems like a bug.

**Comment by gramulos** [comment]
Also, have the same issue.

**Comment by floriancargoet** [comment]
I have a very similar issue:
```js
class SomeComponent extends PureComponent<Props> {
    method({ param }: { param: boolean }) {
        console.log(param, this);
    }

    render() {
        return null;
    }
}
```
```
'value' PropType is defined but prop is never used  react/no-unused-prop-types
```

The issue disappears if I don't spread the parameters `method(dontSpread: { param: boolean }) {`

**Comment by ljharb** [comment]
I think you mean destructure; spread is three dots.

**Comment by floriancargoet** [comment]
You're right, I totally meant destructure. I've edited my message.

**Comment by FerreiraRaphael** [comment]
Have the same issue here:

```javascript
const Wrapper = featureToggle
? ({ children }: { children: Node }) => (
  <FeatureToggledComponent
      featureToggle={featureToggle}
      defaultValue
  >
      {children}
  </FeatureToggledComponent>
)
: React.Fragment;
```

````
75:54  error  'children' PropType is defined but prop is never used  react/no-unused-prop-types
````
